lib/api/model/model.dart

```diff
@@ -178,6 +178,43 @@ enum Emojiset {
     .map((key, value) => MapEntry(value, key));
 }
 
+sealed class MentionableUser {}
+
+class Wildcard extends MentionableUser {
+  Wildcard({
+    required this.name,
+    required this.value,
+    required this.fullDisplayName,
+    required this.type,
+    required this.index,
+  });
+
+  /// The name of the wildcard to be shown as part of [fullDisplayName] in autocomplete suggestions.
+  ///
+  /// Ex: "channel", "stream", "topic", ...
+  final String name;
+
+  /// The value to be put at the compose box after choosing an option from autocomplete.
+  ///
+  /// The same as [name], except for "stream" it is "channel" in FL >= 247 (server-9).
+  final String value; // TODO(sever-9): remove, instead use [name]
+
+  /// The full name of the wildcard to be shown in autocomplete suggestions.
+  ///
+  /// Ex: "all (Notify channel)" or "everyone (Notify recipients)".
+  final String fullDisplayName;
```

Comment:
This seems more about our UI than about the Zulip server API. So let's find a home for it outside of lib/api/.

I think a good solution would look similar to `MentionAutocompleteResult` and its subclasses. Instead of a `User` being an object that knows how to behave as an @-mention autocomplete candidate, we'd have an object that describes an @-mention autocomplete candidate and refers to a `User` to do so.

Reply:
In fact probably a good solution would be for these "autocomplete candidate" objects to be the same objects as the result objects.

I'll try doing a quick refactor in that direction.

Reply:
(→ https://github.com/zulip/zulip-flutter/pull/889#issuecomment-2297696566 below)

Reply:
> This seems more about our UI than about the Zulip server API. So let's find a home for it outside of lib/api/.

Now that the model class is moved outside of lib/api, is it okay for `fullDisplayName` to still live in the class?

Reply:
Yeah. Try to find the arrangement that seems cleanest (and once the PR is ready, reviewers may have suggestions), but between lib/model/ and lib/widgets/ either way is basically fine for where that lives.